smartsim/_core/_cli/scripts/dragon_install.py

```diff
@@ -0,0 +1,273 @@
+    when run on a Cray EX platform
+
+    :param assets: The collection of dragon release assets to filter
+    :returns: An asset meeting platform & version filtering requirements"""
+    # Expect cray & non-cray assets that require a filter, e.g.
+    # 'dragon-0.8-py3.9.4.1-bafaa887f.tar.gz',
+    # 'dragon-0.8-py3.9.4.1-CRAYEX-ac132fe95.tar.gz'
+    asset = next(
+        (
+            asset
+            for asset in assets
+            if _version_filter(asset.name)
+            and _platform_filter(asset.name)
+            and _pin_filter(asset.name)
+        ),
+        None,
+    )
+    return asset
+
+
+def retrieve_asset_info() -> GitReleaseAsset:
+    """Find a release asset that meets all necessary filtering criteria
+
+    :param dragon_pin: identify the dragon version to install (e.g. dragon-0.8)
+    :returns: A GitHub release asset"""
+    assets = _get_release_assets()
+    asset = filter_assets(assets)
+    if asset is None:
+        raise SmartSimCLIActionCancelled("No dragon runtime asset available to install")
+
+    logger.debug(f"Retrieved asset metadata: {asset}")
+    return asset
+
+
+def retrieve_asset(working_dir: pathlib.Path, asset: GitReleaseAsset) -> pathlib.Path:
+    """Retrieve the physical file associated to a given GitHub release asset
+
+    :param working_dir: location in file system where assets should be written
+    :param asset: GitHub release asset to retrieve
+    :returns: path to the downloaded asset"""
+    if working_dir.exists() and list(working_dir.rglob("*.whl")):
+        return working_dir
+
+    archive = WebTGZ(asset.browser_download_url)
+    archive.extract(working_dir)
+
+    logger.debug(f"Retrieved {asset.browser_download_url} to {working_dir}")
+    return working_dir
+
+
+def install_package(asset_dir: pathlib.Path) -> int:
+    """Install the package found in `asset_dir` into the current python environment
+
+    :param asset_dir: path to a decompressed archive contents for a release asset"""
+    package_path = next(asset_dir.rglob("*.whl"), None)
+    if not package_path:
+        logger.error(f"No wheel found for package in {asset_dir}")
+        return 1
+
+    logger.info(f"Installing dragon from: {package_path.absolute()}")
+
+    try:
+        pip("install", "--force-reinstall", str(package_path))
+        return 0
+    except Exception:
+        logger.error(f"Unable to install from {asset_dir}")
+        return 1
+
+
+def cleanup(
+    archive_path: t.Optional[pathlib.Path] = None,
+    asset_dir: t.Optional[pathlib.Path] = None,
+) -> None:
+    """Delete the downloaded asset and any files extracted during installation
+
+    :param archive_path: path to a downloaded archive for a release asset
+    :param asset_dir: path to a decompressed archive contents for a release asset"""
+    if archive_path:
+        archive_path.unlink(missing_ok=True)
+        logger.debug(f"Deleted archive: {archive_path}")
+    if asset_dir:
```

Comment:
per @al-rigazzi, the extracted files are required at runtime and should _not_ be removed.

Reply:
Fixed.